src/interactions/panZoomInteraction.ts

```diff
@@ -20,56 +29,140 @@ export module Interaction {
-        this._zoom.y((<any> this._yScale)._d3Scale);
-      }
-      this._zoom.on("zoom", () => this._rerenderZoomed());
-      this._zoom(this._hitBox);
+      this._dragInteraction = new Interaction.Drag();
+      this._setupDragInteraction();
+      this._touchIds = d3.map();
     }
 
     public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
       super._anchor(component, hitBox);
-      this.resetZoom();
+      this._dragInteraction._anchor(component, hitBox);
+
+      var mouseDispatcher = Dispatcher.Mouse.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      mouseDispatcher.onWheel("Interaction.PanZoom" + this.getID(), (p: Point, e: WheelEvent) => this._handleWheelEvent(p, e));
+
+      this._touchDispatcher = Dispatcher.Touch.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      this._touchDispatcher.onTouchStart("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchStart(ids, idToPoint, e));
+      this._touchDispatcher.onTouchMove("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchMove(ids, idToPoint, e));
+      this._touchDispatcher.onTouchEnd("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchFinish(ids, idToPoint, e));
+      this._touchDispatcher.onTouchCancel("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handlePinchFinish(ids, idToPoint, e));
     }
 
-    public _requiresHitbox() {
-      return true;
+    private _handlePinchStart(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      ids.forEach((id) => {
+        if (this._touchIds.size() === 2) {
+          return;
+        }
+
+        this._touchIds.set(id.toString(), this._translateToComponentSpace(idToPoint[id]));
+      });
     }
 
-    private _rerenderZoomed() {
-      // HACKHACK since the d3.zoom.x modifies d3 scales and not our TS scales, and the TS scales have the
-      // event listener machinery, let's grab the domain out of the d3 scale and pipe it back into the TS scale
-      if (this._xScale) {
-        var xDomain = (<any> this._xScale)._d3Scale.domain();
-        this._xScale.domain(xDomain);
+    private _handlePinchMove(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      if (this._touchIds.size() < 2) {
+        return;
       }
+      var points = this._touchIds.values();
+      var firstTouchPoint = points[0];
+      var secondTouchPoint = points[1];
+
+      var leftX = Math.min(firstTouchPoint.x, secondTouchPoint.x);
+      var rightX = Math.max(firstTouchPoint.x, secondTouchPoint.x);
+      var topY = Math.min(firstTouchPoint.y, secondTouchPoint.y);
+      var bottomY = Math.max(firstTouchPoint.y, secondTouchPoint.y);
+
+      var oldAvgX = (leftX + rightX) / 2;
+      var oldAvgY = (bottomY + topY) / 2;
+      var oldDiff = Math.sqrt(Math.pow(rightX - leftX, 2) + Math.pow(bottomY - topY, 2));
```

Comment:
Is there a way to indicate that the preceding logic applies to the "old" points? Maybe factor the average and diff calculations out into a helper method?